### PR TITLE
feat: can control whether dashboard template is public or not

### PR DESCRIPTION
**Pull request by lharries**
Adds `scope` to control whether a dashboard is visible to everyone or just that team. And defaults it to be team only

This allows the posthog team to create a dashboard template internally first, test it and then share it

## How did you test this code?
Wrote tests for the backend

Tested it manually with the frontend

### CODE REVIEW

File: posthog/models/dashboard_templates.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ class DashboardTemplate(UUIDModel):
     created_by: models.ForeignKey = models.ForeignKey("User", on_delete=models.SET_NULL, null=True, blank=True)
     deleted: models.BooleanField = models.BooleanField(blank=True, null=True)
     image_url: models.CharField = models.CharField(max_length=8201, null=True, blank=True)
+    is_public: models.BooleanField = models.BooleanField(blank=True, null=True)
```
Comment (pauldambra) [inline posthog/models/dashboard_templates.py:28]: If we use a boolean flag then we only have the two states everyone and only the owning team of the instance

Should we have a scope...? Starts as "PUBLIC" or "TEAM" but allows extension to include e.g. "ORGANIZATION"

Comment (lharries) [reply posthog/models/dashboard_templates.py:28]: Yep makes a lot of sense, will switch it

File: frontend/src/types.ts [hunk 2]
```diff
@@ -1234,6 +1234,7 @@ export interface DashboardTemplateType {
     variables?: DashboardTemplateVariableType[]
     tags?: string[]
     image_url?: string
+    scope?: 'team' | 'everyone'
```
Comment (pauldambra) [inline frontend/src/types.ts:1237]: I'm tempted to say "global" is better than everyone... but I'm also worried I'm nit-picking 🤣

**Review by pauldambra — APPROVED** [review]
The only irreversible choice here is the word "everyone" in the DB

I think I'm nitpicking over that so other than a few thoughts that could be altered in follow-up. I've tested this locally and all looks good

File: frontend/src/scenes/dashboard/dashboards/templates/DashboardTemplatesTable.tsx [hunk 3]
```diff
@@ -35,10 +35,10 @@ export const DashboardTemplatesTable = (): JSX.Element | null => {
...
-            title: 'Source',
+            title: 'Type',
             dataIndex: 'team_id',
-            render: (_, { team_id }) => {
-                if (team_id === null) {
+            render: (_, { scope }) => {
+                if (scope === 'everyone') {
                     return <LemonSnack>Official</LemonSnack>
                 } else {
                     return <LemonSnack>Team</LemonSnack>
```
[the remark this answers is not in the document]
Comment (lharries) [reply frontend/src/scenes/dashboard/dashboards/templates/DashboardTemplatesTable.tsx:44]: The only global templates at the start will be by PostHog as so they will be "Official" - we'll need to update this stuff later however if we do community global templates

**Pull request merged**